src/mokka/src/integration-test/java/pl/hycom/mokka/handler/impl/GroovyMockHandlerTest.java

```diff
@@ -0,0 +1,93 @@
+package pl.hycom.mokka.handler.impl;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.mock.web.DelegatingServletInputStream;
+import pl.hycom.mokka.emulator.mock.MockContext;
+import pl.hycom.mokka.emulator.mock.handler.MockHandlerException;
+import pl.hycom.mokka.emulator.mock.handler.impl.GroovyMockHandler;
+import pl.hycom.mokka.emulator.mock.model.GroovyConfigurationContent;
+import pl.hycom.mokka.emulator.mock.model.MockConfiguration;
+import wiremock.org.apache.commons.io.IOUtils;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Collections;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Bartosz Kuron (bartosz.kuron@hycom.pl)
+ */
+public class GroovyMockHandlerTest {
+
+    private static final String FROM = "000.000.0.000";
+    private static final String URL_METHOD = "test-body";
+    private static final String HTTP_METHOD = "GET";
+    private static final String REQUEST_BODY = "requestBody";
+    private static final String SEPARATOR = "result+='|'\n";
+
+    private GroovyMockHandler groovyMockHandler = new GroovyMockHandler();
+    private HttpServletRequest request;
+
+
+    @BeforeEach
+    public void initialize() throws IOException {
+        request = mock(HttpServletRequest.class);
+        when(request.getMethod()).thenReturn(HTTP_METHOD);
+        when(request.getHeaderNames()).thenReturn(Collections.enumeration(new ArrayList<String>(Collections.singletonList("HeaderName"))));
+        when(request.getHeaders(any())).thenReturn(Collections.enumeration(new ArrayList<String>(Collections.singletonList("Header"))));
+        when(request.getAttribute(RequestDispatcher.FORWARD_REQUEST_URI)).thenReturn("/" + URL_METHOD);
+        when(request.getRemoteHost()).thenReturn(FROM);
+        when(request.getInputStream()).thenReturn(new DelegatingServletInputStream(IOUtils.toInputStream(REQUEST_BODY)));
+    }
+
+    @Test
+    public void handle_test() throws IOException, MockHandlerException {
```

Comment:
could you please cover other cases like: null/empty body or text body (meaning: not a groovy script).
In such cases we should return original body